### PR TITLE
Add support for IRC server password

### LINKED ISSUE #8: Does not support IRC server password
I run a personal IRC server with a server password. Currently the 'appservicebot' is able to connect, but virtual users cannot as there's nowhere to specify a password for them.

### PR DESCRIPTION
**Pull request by DeltaWhy**
Fixes #8.

### CODE REVIEW

File: lib/config/schema.yml [hunk 1]
```diff
@@ -82,6 +82,8 @@ properties:
                             type: "boolean"
                         sslselfsign:
                             type: "boolean"
+                        password:
+                            type: "string"
```
Comment (Kegsay) [inline lib/config/schema.yml]: Can you put a small summary of what this does in `config.sample.yaml` please.

**Comment by Kegsay** [comment]
LGTM after comment is done.

**Comment by DeltaWhy** [comment]
Done.

File: config.sample.yaml [hunk 2]
```diff
@@ -31,6 +31,8 @@ ircService:
       ssl: true
       # Whether or not IRC server is using a self-signed cert or not providing CA Chain
       sslselfsign: false
+      # The server password. Optional.
```
Comment (Kegsay) [inline config.sample.yaml]: Specifically this is the password sent in the `PASS` command - we should explicitly state this e.g.

> The password to send for all clients as a `PASS` command.

**Comment by Kegsay** [comment]
LGTM after wording tweak (sorry!)

**Comment by Kegsay** [comment]
Merging and will make the tweak myself.

**Comment by Kegsay** [comment]
https://github.com/matrix-org/matrix-appservice-irc/commit/2928760488baab77dce7f0c63d5f3a6ebe5f43aa

**Pull request merged**